spec/index.md

```diff
@@ -496,36 +496,32 @@ limited capabilities.
 
 The list of types is:
 
-* Primitive types:
+* Base types:
     * [Boolean] — `true` or `false`
+    * [Address] - Ethereum address
     * [Numeric] — integer
-* Sequence types:
+* Reference types:
     * [Tuple]
     * [Array]
     * [Bytes]
+    * [Map]
 * User-defined types:
     * [Struct]
     * [Enum]
-    * [Contract]
-    * [Event]
 * Function types:
     * [Function]
-* Map types:
-    * [HashMap]
 
 ### 5.1.1.1 Boolean type
 
 The `bool` type is a datatype which can be either `true` or `false`.
 
+Example:
+
 ```rust
 x = true
 ```
 
-### 5.1.1.2. Contract type
-
-An *contract type* is the type denotated by the name of an [`contract` item].
-
```

Comment:
Same here, it means that e.g. `GuestBook` would be a type (assuming it is defined as contract)

https://github.com/ethereum/fe/blob/7cd0b98a31d745f8d2c399553cab5f6f49827be8/compiler/tests/fixtures/guest_book.fe#L3-L15